server/routes/ideas.js

```diff
@@ -1,81 +1,268 @@
+// ======================================================================
 const express = require('express');
+const router = express.Router();
+
 const ideasData = require('../data/ideas');
+
 const { successStatusCodes, badRequestErr } = require('../utils');
 const { authenticateToken } = require('../middleware/auth');
-const {
-	isValidIdeaName,
-	isValidStatus,
-	isValidLookingFor,
-} = require('../utils/ideas');
+const { isValidIdeaName, isValidStatus, isValidLookingFor } = require('../utils/ideas');
 const { isValidTechnologies } = require('../utils/projects');
 const { sendErrResp, isValidStr, isValidObjectId } = require('../utils');
 const { isValidUsername } = require('../utils/users');
 
-const router = express.Router();
 
-router.route('/').post(authenticateToken, async (req, res) => {
-	const { user } = req;
-	let { name, description, lookingFor, status, technologies } = req.body;
-	try {
-		user._id = isValidObjectId(user._id);
-		user.username = isValidUsername(user.username);
-		name = isValidIdeaName(name);
-		description = req.body.description
-			? isValidStr(req.body.description, 'idea description')
-			: null;
-		technologies = isValidTechnologies(technologies);
-		lookingFor = isValidLookingFor(lookingFor);
-		status = isValidStatus(status);
-
-		const ideaObject = {
-			name,
-			description,
-			lookingFor,
-			status,
-			technologies,
-		};
-		const idea = await ideasData.createIdea(ideaObject, user);
-		res.status(successStatusCodes.CREATED).json({
-			idea,
-		});
-	} catch (e) {
-		sendErrResp(res, e);
-	}
-});
+// Create a new Idea
+// ======================================================================
+router
+	.route('/')
+	.post(authenticateToken, async (request, response) => {
+		const { user } = request;
+		let { name, description, lookingFor, status, technologies } = request.body;
+		try {
+			// User Validation
+			user._id = isValidObjectId(user._id);
+			user.username = isValidUsername(user.username);
+
+			// Name Validation
+			name = isValidIdeaName(name);
+
+			// Description Validation
+			description = request.body.description
+				? isValidStr(request.body.description, 'idea description')
+				: null;
+
+			// LookingFor Validation
+			lookingFor = isValidLookingFor(lookingFor);
+
+			// Status Validation
+			status = isValidStatus(status);
 
+			// Technologies Validation
+			technologies = isValidTechnologies(technologies);
+
+			const ideaObject = {
+				name,
+				description,
+				lookingFor,
+				status,
+				technologies,
+			};
+
+			const idea = await ideasData.createIdea(ideaObject, user);
+
+			response
+				.status(successStatusCodes.CREATED)
+				.json({
+					idea
+				});
+
+		} catch (e) {
+			sendErrResp(response, e);
+		}
+	});
+
```

Comment:
Just for the sake of adding comments you could have just changed where it was needed.